server/src/graql/query/Query.java

```diff
@@ -95,7 +95,30 @@ public String toString() {
     enum Operator {
         AND("and"),
         OR("or"),
-        NOT("not"),
+        NOT("not");
+
+        private final String operator;
+
+        Operator(String operator) {
+            this.operator = operator;
+        }
+
+        @Override
+        public String toString() {
+            return this.operator;
+        }
+
+        public static Operator of(String value) {
+            for (Operator c : values()) {
+                if (c.operator.equals(value)) {
+                    return c;
+                }
+            }
+            return null;
+        }
+    }
+
+    enum Comparator {
```

Comment:
why don't we make it a class? we would avoid all the procedural code and multiple ifs in ValueExecutor

Reply:
The actual comparison class is in `(ValueProperty|ValueExecutor).Operation.Comparison`. These comparators you're referring to is just the syntax token.